### DISCUSSION TITLE
Scalar array serialisation

### DISCUSSION THREAD

**Discussion by ealeykin** [aaubry/YamlDotNet#652] [Q&A]
Hi,

Is there an options to make the following serialisation:

```
property: [scalar1, scalar2]

instead of

property:
  - scalar1
  - scalar2
```

Thanks

**Comment by EdwardCooke** [comment]
No, the way it is coded is that you would have to build your own emitter to output it that way.

This is the relevant lines in the current emitter implementation that you would want to re-work.

https://github.com/aaubry/YamlDotNet/blob/cda8ccd9ce061093ecf7fd2bee2707c1008aa1dc/YamlDotNet/Core/Emitter.cs#L1570-L1588

**Reply by vhatsura** [reply]
@halamah, you also can create custom implementation of `IYamlTypeConverter` with the following `WriteYaml` method (it's type converter for IEnumerable<string>)

```csharp
public void WriteYaml(IEmitter emitter, object? value, Type type)
    {
        if (value == null) return;

        var sequence = (IEnumerable<string>)value;

        emitter.Emit(new SequenceStart(default, default, false, SequenceStyle.Flow));
        foreach (var item in sequence)
        {
            emitter.Emit(new Scalar(default, item));
        }

        emitter.Emit(new SequenceEnd());
    }
```

the idea is to specify `SequenceStyle`.
